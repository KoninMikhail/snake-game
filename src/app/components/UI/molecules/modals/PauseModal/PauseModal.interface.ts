export interface IPauseModal {
    headline: string;
    message: string;
    onClick?: () => void;
}
