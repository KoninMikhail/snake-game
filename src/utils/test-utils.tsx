import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
