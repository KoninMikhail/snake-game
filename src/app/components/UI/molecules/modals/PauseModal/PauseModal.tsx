import {StyledPauseModal, StyledPauseModalContainer,} from '@ui/molecules/modals/PauseModal/style/StyledPauseModal';
import {ExtraLargeCaption} from '@ui/atoms/typography/caption/ExtraLargeCaption/ExtraLargeCaption';
import {IPauseModal} from './PauseModal.interface';
import {Text} from '@ui/atoms/typography/body/Text/Text';

/**
 * @name PauseModal
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use it with board component
 *
 * @param headline - title of that modal
 * @param message - headline text
 */
export const PauseModal = ({ message, headline, onClick }: IPauseModal) => (
    <StyledPauseModal onClick={onClick}>
        <StyledPauseModalContainer>
            <ExtraLargeCaption align={'center'}>{headline.toUpperCase()}</ExtraLargeCaption>
            <Text>{message}</Text>
        </StyledPauseModalContainer>
    </StyledPauseModal>
);
