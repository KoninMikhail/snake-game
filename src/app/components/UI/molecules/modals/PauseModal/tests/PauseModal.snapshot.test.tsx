import {renderWithProviders} from '@utils/test-utils';
import {PauseModal} from '@ui/molecules/modals/PauseModal/PauseModal';

/**
 * Test
 */
describe('PauseModal', () => {
    it('equal snapshot', () => {
        const { container } = renderWithProviders(
            <PauseModal message={'message'} headline={'headline'} />
        );
        expect(container).toMatchSnapshot();
    });
});
