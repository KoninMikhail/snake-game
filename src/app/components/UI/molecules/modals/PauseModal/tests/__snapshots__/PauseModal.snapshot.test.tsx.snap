// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PauseModal equal snapshot 1`] = `
.c0 {
  position: absolute;
  display: grid;
  top: 0;
  left: 0;
  bottom: 0;
  padding: 15px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  z-index: 5000;
  height: 100%;
  background: black;
}

.c1 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1.5rem;
}

.c2 {
  font-size: 2.3em;
  font-weight: normal;
  text-align: center;
}

.c3 {
  color: white;
  font-size: 14px;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-weight: normal;
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.3em;
  }
}

@media (min-width:1200px) {
  .c2 {
    font-size: 2em;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 16px;
  }
}

@media (min-width:1200px) {
  .c3 {
    font-size: 1.1vw;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        HEADLINE
      </span>
      <span
        class="c3"
      >
        message
      </span>
    </div>
  </div>
</div>
`;
